articles/quickstart/webapp/nodejs/01-login.md

```diff
@@ -14,4 +16,247 @@ github:
 ---
 <%= include('../_includes/_getting_started', { library: 'Node.js', callback: 'http://localhost:3000/callback' }) %>
 
-<%= include('_includes/_login') %>
+## Configure Node.js to use Auth0
+
+### Install the dependencies
+
+To get started, install the following dependencies.
+
+* [passport](http://www.passportjs.org/) - an authentication middleware for Node.js
+* [passport-auth0](https://github.com/auth0/passport-auth0) - an Auth0 authentication strategy for Passport
+* [express-session](https://www.npmjs.com/package/express-session) - a middleware to manage sessions
+* [connect-ensure-login](https://github.com/jaredhanson/connect-ensure-login) - a middleware to ensure a user must be logged in, in order to access certain routes
+
+```bash
+# installation with npm
+npm install passport passport-auth0 express-session connect-ensure-login --save
+```
+
+or
+
+```bash
+# installation with yarn
+yarn add passport passport-auth0 express-session connect-ensure-login
+```
+
+### Configure express-session
+
+In `app.js`, include the `express-session` module and configure it. The `secret` parameter is a secret string that is used to sign the session ID cookie. Please use a custom value.
+
+```js
+// app.js
+
+var session = require('express-session');
+
+//session-related stuff
+var sess = {
+ secret: 'CHANGE THIS SECRET',
+ cookie: {},
+ resave: false,
+ saveUninitialized: true
+};
+
+if (app.get('env') === 'production') {
+ sess.cookie.secure = true; // serve secure cookies, requires https
+}
+
+app.use(session(sess));
+```
+
+### Configure Passport with the application settings
+
+In `app.js`, include the `passport` and `passport0-auth0` modules, and configure Passport to use a new instance of `Auth0Strategy` with your Auth0 application settings. Use `passport.initialize()` and `passport.session()` to initialize Passport with persistent login sessions.
+
+```js
+// app.js
+var Auth0Strategy = require('passport-auth0'),
+    passport = require('passport');
+
+//passport-auth0
+var strategy = new Auth0Strategy({
+  domain: '${account.namespace}',
```

Comment:
Another note - `${account.namespace}` will grab the tenant URL such as `foo.auth0.com`. However, if the customer is using a Custom Domain (`login.foo.com`) this URL should probably be that one, right? If so, we should probably communicate that somehow here, but without getting bogged down in those details too much.